worker/s3caller/manifold_test.go

```diff
@@ -42,9 +39,8 @@ func (s *manifoldSuite) TestValidateConfig(c *gc.C) {
 
 func (s *manifoldSuite) getConfig() ManifoldConfig {
 	return ManifoldConfig{
-		APIConfigWatcherName: "api-config-watcher",
-		APICallerName:        "api-caller",
-		NewS3Client: func(api.Connection, Logger) (Session, error) {
+		APICallerName: "api-caller",
+		NewClient: func(api.Connection, Logger) (objectstore.Session, error) {
```

Comment:
I'd argue that returning an instance type like `objectsClient` (we should make it public though) is better than importing an interface from core only as return type, right?

Reply:
We don't want to bring worker package into other packages. One day I'll move all the workers into the internal package, just not today.

With this in mind, we need to put the types into core so others can depend on them. We've done this for Tracing, DBAccessor, ChangeStream. It's a common pattern we're building upon.